python/pyspark/ml/linalg/__init__.py

```diff
@@ -980,14 +980,14 @@ def __getitem__(self, indices):
             return self.values[i + j * self.numRows]
 
     def __eq__(self, other):
-        if (not isinstance(other, DenseMatrix) or
-                self.numRows != other.numRows or
-                self.numCols != other.numCols):
+        if isinstance(other, SparseMatrix):
+            return np.all(self.toArray() == other.toArray())
+        if (self.numRows != other.numRows or self.numCols != other.numCols):
```

Comment:
I think it would make sense to move this check above L983? We canm check the number of rows and columns faster than converting a `SparseMatrix` to an array.